eth/eth_test.go

```diff
@@ -128,10 +128,11 @@ func TestGetBlockByHash(t *testing.T) {
 
 func TestGetProof(t *testing.T) {
 	someAddress := common.HexToAddress("0xabc")
+	blockstore := store.NewBlockStore(testutils.NewMemDB(t)) // this blockstore contains no blocks.
 
-	proofProvider := eth.NewProofProvider()
+	proofProvider := eth.NewProofProvider(nil, blockstore)
 
 	pf, err := proofProvider.GetProof(someAddress, []string{}, nil)
-	require.ErrorIs(t, err, eth.ErrNotImplemented, "getProof didn't fail")
-	require.Nil(t, pf, "received proof")
+	require.Error(t, err, "should not succeed in generating proofs with empty blockstore")
+	require.Nil(t, pf, "received proof from empty blockstore")
```

Comment:
This is fine for now, I agree that we'll want to add more comprehensive unit tests for this later on though. Should I create a subtask in the withdrawals page for adding this test or did you already have something planned?

Reply:
Could create a subtask. I mostly wanted to checkpoint before investing much time in creating the "wrong" types of testdata.